src/adapter/src/rbac.rs

```diff
@@ -146,6 +151,86 @@ pub fn check_item_usage(
     Ok(())
 }
 
+fn requires_item_usage_privileges(plan: &Plan) -> bool {
+    match plan {
+        Plan::CreateConnection(_)
+        | Plan::CreateDatabase(_)
+        | Plan::CreateSchema(_)
+        | Plan::CreateRole(_)
+        | Plan::CreateCluster(_)
+        | Plan::CreateClusterReplica(_)
+        | Plan::CreateSource(_)
+        | Plan::CreateSources(_)
+        | Plan::CreateSecret(_)
+        | Plan::CreateSink(_)
+        | Plan::CreateTable(_)
+        | Plan::CreateView(_)
+        | Plan::CreateMaterializedView(_)
+        | Plan::CreateIndex(_)
+        | Plan::CreateType(_)
+        | Plan::Comment(_)
+        | Plan::DiscardTemp
+        | Plan::DiscardAll
+        | Plan::DropObjects(_)
+        | Plan::DropOwned(_)
+        | Plan::EmptyQuery
+        | Plan::ShowAllVariables
+        | Plan::ShowColumns(_)
+        | Plan::ShowVariable(_)
+        | Plan::InspectShard(_)
+        | Plan::SetVariable(_)
+        | Plan::ResetVariable(_)
+        | Plan::SetTransaction(_)
+        | Plan::StartTransaction(_)
+        | Plan::CommitTransaction(_)
+        | Plan::AbortTransaction(_)
+        | Plan::Select(_)
+        | Plan::Subscribe(_)
+        | Plan::CopyFrom(_)
+        | Plan::ExplainTimestamp(_)
+        | Plan::Insert(_)
+        | Plan::AlterCluster(_)
+        | Plan::AlterNoop(_)
+        | Plan::AlterIndexSetOptions(_)
+        | Plan::AlterIndexResetOptions(_)
+        | Plan::AlterSetCluster(_)
+        | Plan::AlterSink(_)
+        | Plan::AlterSource(_)
+        | Plan::PurifiedAlterSource { .. }
+        | Plan::AlterClusterRename(_)
+        | Plan::AlterClusterReplicaRename(_)
+        | Plan::AlterItemRename(_)
+        | Plan::AlterSecret(_)
+        | Plan::AlterSystemSet(_)
+        | Plan::AlterSystemReset(_)
+        | Plan::AlterSystemResetAll(_)
+        | Plan::AlterRole(_)
+        | Plan::AlterOwner(_)
+        | Plan::Declare(_)
+        | Plan::Fetch(_)
+        | Plan::Close(_)
+        | Plan::ReadThenWrite(_)
+        | Plan::Prepare(_)
+        | Plan::Execute(_)
+        | Plan::Deallocate(_)
+        | Plan::Raise(_)
+        | Plan::RotateKeys(_)
+        | Plan::GrantRole(_)
+        | Plan::RevokeRole(_)
+        | Plan::GrantPrivileges(_)
+        | Plan::RevokePrivileges(_)
+        | Plan::AlterDefaultPrivileges(_)
+        | Plan::ReassignOwned(_)
+        | Plan::SideEffectingFunc(_)
+        | Plan::ValidateConnection(_) => true,
+        Plan::ShowCreate(_) => false,
+        Plan::ExplainPlan(plan) => match plan.explainee {
+            Explainee::MaterializedView(_) | Explainee::Index(_) => false,
```

Comment:
This is false because having USAGE on a schema means that user should be able to EXPLAIN a mv/index? Can this get a test too?

Reply:
Yeah. I was a bit pedantic with this case, `true` or `false` both would had the same behavior because materialized views and indexes don't have usage privileges. However, it's more of a hint that if we ever add something like `EXPLAIN CONNECTION c` it must be `false`.

I'll add a test.

Reply:
I just added the test (and in the process found a small related bug that I fixed).